pset-iterators.js

```diff
@@ -171,6 +243,33 @@ console.log('Problem 6:')
 
 // Add your code below this line
 
+const shortTermPresidents = function(){
+
+  // remove the president who hasn't left office yet
+  presidents.pop()
+
+  // create a new array of presidental objects with added integer properties
+  const presidentsWithAddedProperties = presidents.map(function(president){
+    president.took_office_year = parseInt(president.took_office.substring(0, 4))
+    president.left_office_year = parseInt(president.left_office.substring(0, 4))
+    return president
+  })
+
+  // filter out presidents who served less than four years
+  const presidentsWhoServedLessThanFourYears = presidentsWithAddedProperties.filter(function(president){
+    return president.left_office_year - president.took_office_year < 4
+  })
```

Comment:
@carly-ga this works fine but an alternative approach would have been to skip the step of adding properties to the presidents array and just handle all of that logic in the `.filter()` iterator

```
// formerPresidents is an array of all presidents except the current one
const shortTermPresidents = formerPresidents.filter((president) => {
  const tookOfficeYear = parseInt(president.took_office.substring(0, 4))
  const leftOfficeYear = parseInt(president.left_office.substring(0, 4))

  return leftOfficeYear - tookOfficeYear < 4
})
```